torch/_decomp/decompositions_for_jvp.py

```diff
@@ -179,7 +179,7 @@ def native_layer_norm_backward(
         if len(outer_dim_indices) > 0:
             d_bias: Optional[Tensor] = torch.sum(grad_out, outer_dim_indices, False)
         else:
-            d_bias = grad_out
+            d_bias = grad_out.view_as(grad_out)
```

Comment:
I have two questions:
1. `native_layer_norm_backward`'s decomposition only gets registered to `decomposition_table_for_jvp`. Is there a reason why we didn't put it into torch._decomps.decomposition_table? @Chillee 
2. Claim: a decomposition for an operator should match the alias information in the schema. Because the alias information for native_layer_norm_backward explicitly says that `grad_out` is not aliased, then we should be returning a clone here, otherwise we are lying.

Reply:
> native_layer_norm_backward's decomposition only gets registered to decomposition_table_for_jvp. Is there a reason why we didn't put it into torch._decomps.decomposition_table? @Chillee

This one is from me--(tl;dr, it's the norms issue). So there is a version of this that's basically the same in the `torch._decomps.decomposition_table` but the only difference is that this recomputes mean and var.

When running in eager at least, all norms end up in this state where the mean and var passed into the function don't have autograd tracking back to input. They assume that the backwards and double backwards kernels will know about this dependency since the kernels are specific to the norms. However, when we're doing decompositions, since we aren't writing a custom forwards over reverse kernel for the norm, we need to propagate this information to the so we end up recomputing versions of the mean and variance that do depend on the input. I didn't put this in the decompositions table because it really felt hacky. However, not sure if this is something that might come up with the normal decompositions (to be clear, the non-hacky version computes native_layer_norm_backward correctly but just doesn't do the forwards AD over that function correctly)

Reply:
(1) It does some recomputation for the purposes of higher-order gradient computation that we otherwise wouldn't need.
(2) Agreed. `clone` is probably a better choice here.

Reply:
Ahh it's that. Thanks for the explanations